src/app/qgslayercapabilitiesmodel.cpp

```diff
@@ -184,6 +194,10 @@ Qt::ItemFlags QgsLayerCapabilitiesModel::flags( const QModelIndex &idx ) const
     {
       return Qt::ItemIsEnabled | Qt::ItemIsSelectable | Qt::ItemIsEditable | Qt::ItemIsUserCheckable;
     }
+    else if ( idx.column() == HiddenColumn )
```

Comment:
could you turn this into a switch?

Reply:
Sure, it's just funny you are asking  .. https://github.com/qgis/QGIS/commit/6b66b2048a7d06b7db9ae33054fe3b2d5314e571#diff-5e009636e6446c3a89221a5d469cf0619daade026ea52c7cea0abb66888750aeR129

Reply:
sorry, probably because of the first if on top (LayerColumn).
let's move it to the switch condition too, and do `if ( !layer && idx.column() != LayerColumn )`

Reply:
(as for now you'll have a warning since LayerColumn is not handled in the switch)

Reply:
no warnings, because there is no cast, I'll add it.